components/treelist/state.md

```diff
@@ -4,9 +4,875 @@ page_title: TreeList - State
+
+    // data generation
+
+    async Task<List<Employee>> GetTreeListData()
+    {
+        List<Employee> data = new List<Employee>();
+
+        for (int i = 1; i < 15; i++)
+        {
+            data.Add(new Employee
+            {
+                Id = i,
+                ParentId = null, // indicates a root-level item
+                Name = $"root: {i}",
+                EmailAddress = $"{i}@example.com",
+                HireDate = DateTime.Now.AddYears(-i)
+            }); ;
+
+            for (int j = 1; j < 4; j++)
+            {
+                int currId = i * 100 + j;
+                data.Add(new Employee
+                {
+                    Id = currId,
+                    ParentId = i,
+                    Name = $"first level child {j} of {i}",
+                    EmailAddress = $"{currId}@example.com",
+                    HireDate = DateTime.Now.AddDays(-currId)
+                });
+
+                for (int k = 1; k < 3; k++)
+                {
+                    int nestedId = currId * 1000 + k;
+                    data.Add(new Employee
+                    {
+                        Id = nestedId,
+                        ParentId = currId,
+                        Name = $"second level child {k} of {i} and {currId}",
+                        EmailAddress = $"{nestedId}@example.com",
+                        HireDate = DateTime.Now.AddMinutes(-nestedId)
+                    }); ;
+                }
+            }
+        }
+
+        return await Task.FromResult(data);
+    }
+}
+````
+````Service
+using System.Threading.Tasks;
+using Microsoft.JSInterop;
+using System.Text.Json;
+
+public class LocalStorage
+{
+    protected IJSRuntime JSRuntimeInstance { get; set; }
+
+    public LocalStorage(IJSRuntime jsRuntime)
+    {
+        JSRuntimeInstance = jsRuntime;
+    }
+
+    public ValueTask SetItem(string key, object data)
+    {
+        return JSRuntimeInstance.InvokeVoidAsync(
+            "localStorage.setItem",
+            new object[] {
+                key,
+                JsonSerializer.Serialize(data)
+            });
+    }
+
+    public async Task<T> GetItem<T>(string key)
+    {
+        var data = await JSRuntimeInstance.InvokeAsync<string>("localStorage.getItem", key);
+        if (!string.IsNullOrEmpty(data))
+        {
+            return JsonSerializer.Deserialize<T>(data);
+        }
+
+        return default;
+    }
+
+    public ValueTask RemoveItem(string key)
+    {
+        return JSRuntimeInstance.InvokeVoidAsync("localStorage.removeItem", key);
+    }
+}
+````
+
+### Set TreeList Options Through State
+
+The TreeList state allows you to control the behavior of the TreeList programmatically - you can, for example, set sorts, filteres and expand items.
+
+>tip The individual tabs below show how you can use the state to programmatically set the TreeList filtering, sorting and other features.
+
+@[template](/_contentTemplates/treelist/state.md#initial-state)
+
+````ExpandedItems
```

Comment:
If there is time, consider a separate example section about changing the data and controlling which of the new items are expanded ,this is the common use case for this. See more details starting from this thread https://feedback.telerik.com/blazor/1481700-control-the-expanded-collapsed-state-of-treelist-items